pkg/util/containers/filter_test.go

```diff
@@ -700,14 +703,14 @@ func TestParseFilters(t *testing.T) {
 	}
 }
 
-func resetConfig() {
-	pkgconfigsetup.Datadog().SetWithoutSource("exclude_pause_container", true)
-	pkgconfigsetup.Datadog().SetWithoutSource("container_include", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("container_exclude", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("container_include_metrics", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("container_exclude_metrics", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("container_include_logs", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("container_exclude_logs", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("ac_include", []string{})
-	pkgconfigsetup.Datadog().SetWithoutSource("ac_exclude", []string{})
+func resetConfig(cfg pkgconfigmodel.Config) {
+	cfg.SetWithoutSource("exclude_pause_container", true)
+	cfg.SetWithoutSource("container_include", []string{})
+	cfg.SetWithoutSource("container_exclude", []string{})
+	cfg.SetWithoutSource("container_include_metrics", []string{})
+	cfg.SetWithoutSource("container_exclude_metrics", []string{})
+	cfg.SetWithoutSource("container_include_logs", []string{})
+	cfg.SetWithoutSource("container_exclude_logs", []string{})
+	cfg.SetWithoutSource("ac_include", []string{})
+	cfg.SetWithoutSource("ac_exclude", []string{})
```

Comment:
This functions aims at reverting the config changes that used to be done on the config singleton.
Now that we have a mock config object, it should be smarter to just drop the mock config object and re-create a fresh one rather trying to reset one.

Reply:
Because the mock config caches its reference to the global config, it's not possible to reset its state by making a new local object. So this function to manually reset the state is still required. Arguably, this test could be split into smaller individual tests, but that's a larger change than I'd like to make in a PR that's already touching so many files.